### PR TITLE
Making OrbitService compile without OpenGL

### PR DESCRIPTION
**Pull request by beckerhe**
That's a somehow hacky commit and just intended as a temporary solution.

In parallel to OrbitGl, this PR introduces another cmake target, called OrbitNoGl.
OrbitNoGl derives from OrbitGl in that sense, that it compiles the same cpp-files
(except some pure UI related cpp files) but with an additional preprocessor
definition called `NOGL`.

Additionally I added ifdef-guards to all calls into opengl-related code. These
preprocessor checks enable the calls in the absence of `NOGL`.

I tested OrbitService on a Stadia instance with Unreal's Infiltrator Demo:
- Connecting to the instance works
- Sampling works
- Hooking functions works

I haven't done any further tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,8 @@ endif()
 project(Orbit C CXX)
 enable_testing()
 
+option(WITH_UI "Setting this option will enable the Qt-based UI client." ON)
```
Comment (ronaldfw) [inline CMakeLists.txt]: Nit: GUI instead of UI?

File: OrbitGl/App.cpp [hunk 2]
```diff
@@ -17,14 +17,20 @@
 #include "Callstack.h"
 #include "Capture.h"
 #include "CaptureSerializer.h"
+#ifndef NOGL
```
Comment (ronaldfw) [inline OrbitGl/App.cpp]: The double negative here and below is a little harder to parse than something like #ifdef GL. Would it make sense to change it or does that cause other weirdnesses elsewhere? I understand you want default behavior when nothing is defined explicitly, so probably better to leave as-is.

Comment (beckerhe) [reply OrbitGl/App.cpp]: Yes, that was the intention. Not to change the default behaviour. So I would prefer to leave it like so.

File: CMakeLists.txt (hunk 1 above)
Comment (beckerhe) [reply CMakeLists.txt]: Yes, sounds better. I'm gonna change that. Hold on...

File: OrbitGl/App.cpp (hunk 2 above)
Comment (ronaldfw) [reply OrbitGl/App.cpp]: Ok, sounds good!

**Comment by beckerhe** [comment]
> Thanks, overall looks good to me and thanks for jumping on this quickly!
> 
> Could you add a comment what kind of testing you have done? And since this is temporary, could you add an issue to make sure cleaning this up is tracked?

I added a comment to the CMakeLists.txt file, where this new target is defined.
I had the feeling, this is where all the strings come together and is probably the
best place to put such a comment.

**Pull request merged**